### PR TITLE
Fixes #52: Sorting can be used in other subclasses

### LINKED ISSUE #52: Move sorting stuff form Admin::BaseController to ApplicationController
It's possible to require this functionality in other namespaces...ya dingus

### PR DESCRIPTION
**Pull request by jordanmaguire**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -44,6 +48,18 @@ def infer_params_key(resource)
...
+# Sorting
+
+  # Sort by :id by default, so there is a consistent order even when no order has been
+  # selected via the interface
+  def default_sort_options
+    {id: :desc}
+  end
+
+  def sort(collection)
+    ModelSorter.sort(collection, params, default_sort_options)
```
Comment (osiro) [inline app/controllers/application_controller.rb]: Suggestion:

```ruby
model_class = "#{collection.klass}Sorter".constantize rescue ModelSorter
model_class.sort(collection, params, default_sort_options)
```

So whenever you need to customise a sorter, like sorting by association, all you need to do is something like:

```ruby
class OrderSorter < ModelSorter

private
  def self.sort_scope(scope, attribute, direction)
    case attribute
    when "customer"
      scope.order("customers.name #{direction}")
    else
      super(scope, attribute, direction)
    end
  end
end
```

[PP] I'd just move ModelSorter to /app, but that's my personal preference...

**Comment by osiro** [comment]
:+1:

**Comment by swebb** [comment]
:+1:

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (jordanmaguire) [inline app/controllers/application_controller.rb]: Hey @osiro in that case is there any reason you wouldn't just override the sort method in the subclass?

```ruby
def sort(collection)
  MyCustomSorter.sort(collection, params, default_sort_options)
end
```

Comment (osiro) [inline app/controllers/application_controller.rb]: You can do that too.

We have pretty much 2 options:

1. keeping our controllers the same, and making the ApplicationController::sort infer the sort class
2. override sort method in children controllers

Both seems okay to me.

**Pull request merged**